packages/xml-views-validation/test/validators/element/type-of-aggregation-spec.ts

```diff
@@ -0,0 +1,97 @@
+import { partial } from "lodash";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { generateModel } from "@ui5-language-assistant/test-utils";
+import {
+  assertNoIssues as assertNoIssuesBase,
+  assertSingleIssue as assertSingleIssueBase,
+} from "../../test-utils";
+import { validateAggregationType } from "../../../src/validators/elements/type-of-aggregation";
+import {
+  getMessage,
+  INVALID_AGGREGATION_TYPE,
+} from "../../../src/utils/messages";
+
+describe("the type aggregation validation", () => {
+  let ui5SemanticModel: UI5SemanticModel;
+
+  before(async () => {
+    ui5SemanticModel = await generateModel({ version: "1.74.0" });
+  });
+
+  context("true positive scenarios", () => {
+    let assertSingleIssue: (xmlSnippet: string, message: string) => void;
+    before(() => {
+      assertSingleIssue = partial(
+        assertSingleIssueBase,
+        ui5SemanticModel,
+        {
+          element: [validateAggregationType],
+        },
+        "InvalidAggregationType",
+        "error"
+      );
+    });
+
+    it("will detect mismatch of class to explicit aggregation type", () => {
+      assertSingleIssue(
+        `<mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+            xmlns:mvc="sap.ui.core.mvc"
+            xmlns="sap.ui.commons">
+            <m:Panel>
+              <m:headerToolbar>
+                <🢂Toolbar🢀></Toolbar>
```

Comment:
ToolBar does not match the ype of headerToolBa?

Reply:
maybe choose something else like a `Button` so its more obvious it should not match?

Reply:
This is not resolved.
The error message itself is misleading:
- `getMessage(INVALID_AGGREGATION_TYPE, "Toolbar", "Toolbar")`

Can we please pick a more logical scenario to fail? e.g some Button inside headerToolbar